### DISCUSSION TITLE
Recommended place to git clone this repo?

### DISCUSSION THREAD

**Discussion by webel** [hlky/stable-diffusion-webui#768] [Q&A]
Seeing as the stable-diffusion fork is no longer, where should this repo be cloned into? Nested in the stable diffusion directory? I'm looking at the docker files and they look the same, so looks like these files should be amongst the stable-diffusion files, but that would affect pulling both repos...

**Comment by DasWookie** [comment]
You clone it into it's own location, where ever you want, but the default is "stable-diffusion-webui" and then use the sync_local.py script to copy the files into your SD path, like so:
conda activate ldm
cd stable-diffusion-webui
python sync_local.py --dest ../stable_diffusion

**Reply by micwalk** [reply]
Now that the repos are merged the `sync_local` script is no longer needed. Just clone this repo to its own directory, add the checkpoint(s) and execute from there.

**Reply by DasWookie** [reply]
This has been a problematic migration on my side of things. updating basically broke a working system where I was able to run the UI as well as the txt2img and img2img scripts. Now... command line isn't working (after hunting around for the old files) but I was able to get txt2img_half.py script to work. The UI tho, is maddeningly broken for img2img and the old fix of adding something, anything, to a mask still fails with:
Traceback (most recent call last):
  File "/home/cyber/miniconda3/envs/lsd/lib/python3.8/site-packages/gradio/routes.py", line 247, in run_predict
    output = await app.blocks.process_api(
  File "/home/cyber/miniconda3/envs/lsd/lib/python3.8/site-packages/gradio/blocks.py", line 639, in process_api
    processed_input = self.preprocess_data(fn_index, raw_input, state)
  File "/home/cyber/miniconda3/envs/lsd/lib/python3.8/site-packages/gradio/blocks.py", line 543, in preprocess_data
    processed_input.append(block.preprocess(raw_input[i]))
  File "/home/cyber/miniconda3/envs/lsd/lib/python3.8/site-packages/gradio/components.py", line 1546, in preprocess
    x, mask = x["image"], x["mask"]
TypeError: string indices must be integers

so I don't think img2img is working for anyone with a clean install.

**Reply by webel** [reply]
@DasWookie seems like there was a commit 4h ago that fixes that bug, not sure if it's been merged to main yet thou

**Reply by webel** [reply]
ah I don't dare brake my setup lol, spent forever getting the docker working on my paperspace linux

**Reply by DasWookie** [reply]
The problem is, clean installs aren't working. After the merge, I basically wiped everything (I did keep copies of my ckpt files) and did the clean install... txt2img is working. Doesn't seem to be working as well for faces for some reason for me, but that might just be my test prompts. GFPGAN is at least loading and seems to be there.

I understand the merge was a MASSIVE undertaking, but without testing from an entirely clean system, it's not validating the merge. So, some people are OK, and some aren't... and all NEW people (or wiped clean) seem to be broken. I'm still testing and trying to figure out what's what, but it's frustrating this wasn't tested better. That might be my 25+ years of software testing career speaking tho.

**Reply by webel** [reply]
I hear you, I'm over on automatic1111 now, but having a tonne of issues getting that rolling on linux... ah, I just want to make pretty art

**Reply by DasWookie** [reply]
OK! I've finally gotten it working again in a from scratch environment. I'm on linux as well. I had to modify the following:
entrypoint.sh : Modify the pathing for the sd path. It seems to have been hardcoded for a windows install with \sd\ as the root. Changing the instances of "/sd/" to instead be "./" got that resolved. There were 7 instances that needed to be updated.

webui.sh : I had to paste my conda initialize into the top of the script from my .bashrc to get it working. This might have been something wonky with my miniconda setup, but that was the only way I was able to get it to run. I also changed my environment name from "ldm" to "ldo" so my old ldm or lsd environments wouldn't conflict.

I also had to hand install the gfpgan, latent-diffusion, and realesrgan files. Those are technically optional components, so maybe the auto install shouldn't have been expected and I should have known I needed to install those.
